emission/tests/funcTests/TestFunctionTiming.py

```diff
@@ -35,207 +17,114 @@ def setUpClass(self):
-        - name (str): The name of the function or code block.
-        - expected_elapsed_ms (float): The expected elapsed time in milliseconds.
+        :param expected_entries: A list of tuples which have (key, name, expected_elapsed_ms).
         """
-        logging.info(f"Verifying entry for '{name}' under key '{key}'.")
-        stored_documents_chain = self.timeseries_db.find_entries(
-            [key],
-            extra_query_list=[{"data.name": name}]
-        )
-        stored_documents = list(stored_documents_chain)
-
-        self.assertEqual(
-            len(stored_documents),
-            1,
-            f"Expected exactly 1 entry for '{name}' under key '{key}', found {len(stored_documents)}."
-        )
-        stored_document = stored_documents[0]
-
-        try:
-            stored_reading = stored_document['data']['reading']
-            stored_name = stored_document['data']['name']
-            logging.info(
-                f"Stored Document for '{name}': reading={stored_reading} ms, name={stored_name}"
-            )
-        except KeyError as e:
-            self.fail(
-                f"Missing key {e} in stored document for '{name}'."
-            )
-
-        # Log the expected and actual readings
-        logging.info(
-            f"'{name}': Expected elapsed time = {expected_elapsed_ms:.2f} ms, "
-            f"Stored reading = {stored_reading} ms."
-        )
+        logging.debug(f"Ensuring {len(expected_entries)} entries exist in DB.")
+        key_list = [key for (key, _, _) in expected_entries]
+        stored_entrys = list(self.timeseries_db.find_entries(key_list))
+        self.assertEqual(len(stored_entrys), len(expected_entries))
 
-        # Assert that the stored_reading matches elapsed_ms exactly
-        self.assertEqual(
-            stored_reading,
-            expected_elapsed_ms,
-            msg=(
-                f"Stored reading {stored_reading} ms does not exactly match "
-                f"elapsed time {expected_elapsed_ms:.2f} ms for '{name}'."
-            )
-        )
-
-        # Assert that the stored_name matches name exactly
-        self.assertEqual(
-            stored_name,
-            name,
-            msg=(
-                f"Stored name '{stored_name}' does not match function "
-                f"name '{name}'."
-            )
-        )
+        for i in range(len(expected_entries)):
+            stored_entry = stored_entrys[i]
+            expected_key, expected_name, expected_reading = expected_entries[i]
+            logging.debug(f"Comparing expected {expected_entries[i]} " +
+                          f"with stored {stored_entry['metadata']['key']} {stored_entry['data']}")
+            self.assertEqual(stored_entry['metadata']['key'], expected_key)
+            self.assertEqual(stored_entry['data']['name'], expected_name)
+            self.assertEqual(stored_entry['data']['reading'], expected_reading)
```

Comment:
Validating the stored entries vs. the expected entries boils down to this:
checking that key, name, and reading align with the corresponding expected values at the same index

Reply:
Instead of several verbose log statements, I just have one concise log statement that includes all the expected values and all the relevant stored values

Reply:
You can add comment(s) here if you think it would aid future maintainers in understanding these assertions